### ISSUE TITLE
Flush-ContainerHelperCache unexpectedly removes an image

### ISSUE THREAD

**Issue by gunnargestsson** [microsoft/navcontainerhelper#2864]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
Flush-ContainerHelperCache -cache "images" -keepDays 40
command is being used in the pipeline to clean old images

**Scripts used to create container and cause the issue**
```
Flush-ContainerHelperCache -cache "images" -keepDays 40
```

**Full output of scripts**
```
BcContainerHelper version 4.0.10
Setting bcartifactsCacheFolder = F:\Artifacts.cache
BC.HelperFunctions emits usage statistics telemetry to Microsoft
ci:onprem-19.5.36567.36700-w1 was built on artifacts which was removed from the cache, removing image
Untagged: ci:onprem-19.5.36567.36700-w1
Completed
...
```

**Screenshots**
![image](https://user-images.githubusercontent.com/23558797/213199666-2c9dbbad-3edd-4141-a98a-82cc9327f0d5.png)

**Additional context**
All the time.
The artifacts where last used by the same build as you can see on the schreenshot.

**Comment by gunnargestsson** [comment]
There where 3 images at that time built on active artifacts, but clean command only removed one of the images.

**Comment by freddydk** [comment]
So, you are saying that the image was deleted and the artifacts are still there???

**Comment by gunnargestsson** [comment]
Yes, that is what I am saying!

**Comment by gunnargestsson** [comment]
When I don't specify the cache to clean the images are not deleted.

**Comment by gunnargestsson** [comment]
Is there a way to store the image with test toolkit apps installed?

**Comment by freddydk** [comment]
Shipped in 4.0.14
